app/controllers/projects_controller.rb

```diff
@@ -110,14 +112,36 @@ def require_project
     @project = (Project.find_by_param!(params[:id]) if params[:id])
   end
 
-  def as_csv
-    projects = Project.order(:id).all
+  # Determine the timestamp of the successful deploy for each project, and store
+  # the Datetime in a hash that is indexed by project id
+  def project_last_deployed_at
```

Comment:
If it were a model method, it would do n+1 database queries.  Doing it this way allows me to do a single query that will load them all at once.